sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -241,7 +244,7 @@ class GpuHashAggregateIterator(
       boundInputReferences: Seq[GpuExpression],
       boundFinalProjections: Option[Seq[GpuExpression]],
       boundResultReferences: Seq[Expression],
-      aggModeCudfAggregates: Seq[(AggregateMode, Seq[CudfAggregate])])
+      aggModeCudfAggregates: Seq[(GpuAggregateExpression, AggregateMode, Seq[CudfAggregate])])
```

Comment:
For me any tuple above 2 entries that is not produced and consumed in essentially the same few lines of code needs to be replaced with a case class for readability.

Reply:
Done in: https://github.com/NVIDIA/spark-rapids/pull/3417/commits/d2ff0e17fcee45ec4120b5484854f14b60229984